openedx/core/djangoapps/user_authn/views/tests/test_register.py

```diff
@@ -130,6 +133,22 @@ def create_account_and_fetch_profile(self, host='microsite.example.com'):
         user = User.objects.get(username=self.username)
         return user.profile
 
+    @override_waffle_flag(PASSWORD_UNICODE_NORMALIZE, active=True)
+    def test_create_account_and_normalize_password(self, host='microsite.example.com'):
+        """
+        Test that unicode normalization on passwords is happening when a user registers.
+        """
+
+        self.params['password'] = unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt')
+        response = self.client.post(self.url, self.params, HTTP_HOST=host)
+        self.assertEqual(response.status_code, 200)
+
+        user = User.objects.get(username=self.username)
+        salt_val = user.password.split('$')[1]
+
+        user_hashed_password = make_password(unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'), salt_val)
+        self.assertEqual(user_hashed_password, user.password)
```

Comment:
Super nit: this is a test, so who cares. But I can only easily follow the logic of why we're doing what we are here because I'm steeped in this PR. Comments might help.